llvm/lib/Target/Sparc/MCTargetDesc/SparcAsmBackend.cpp

```diff
@@ -132,14 +132,18 @@ namespace {
   class SparcAsmBackend : public MCAsmBackend {
   protected:
     const Target &TheTarget;
+    const MCSubtargetInfo &TheSTI;
     bool Is64Bit;
+    bool HasV9;
 
   public:
-    SparcAsmBackend(const Target &T)
+    SparcAsmBackend(const Target &T, const MCSubtargetInfo &STI)
         : MCAsmBackend(StringRef(T.getName()) == "sparcel"
                            ? llvm::endianness::little
                            : llvm::endianness::big),
-          TheTarget(T), Is64Bit(StringRef(TheTarget.getName()) == "sparcv9") {}
+          TheTarget(T), TheSTI(STI),
+          Is64Bit(StringRef(TheTarget.getName()) == "sparcv9"),
```

Comment:
Unrelated to your change, but I wonder whether checking for the target name being `sparcv9` is the proper and robust way to check whether we're on 64-bit SPARC. On Linux, the 64-bit SPARC target is called `sparc64` and not `sparcv9`.

Reply:
Now you have STI, you can just get the triple and avoid ugly string comparisons for all these (isArch64Bit and isLittleEndian)

Reply:
Okay, thanks for the clarification. We're good then. ;-)